src/core/libmaven/mzUtils.cpp

```diff
@@ -593,93 +426,90 @@ Series:  Prentice-Hall Series in Automatic Computation
-        return (sumxy -( sumx*sumy)/n) / sqrt((x2-(sumx*sumx)/n)*(y2-(sumy*sumy)/n));
+        float corr = (sumxy -( sumx * sumy) / n) /
+                       sqrt((x2 - (sumx * sumx) / n) *
+                          (y2 - (sumy * sumy) / n));
+        return corr;
     }
 
-
-    /*peak fitting function*/
-    void gaussFit(const vector<float>&ycoord, float* sigma, float* R2) {
-
+    pair<float, float> gaussFit(const vector<float>& ycoord)
+    {
         float s = 20;
         float min_s = 0;
         float minR = 1e99;
 
         //find best fit
-        if (ycoord.size()<3) return;
-        vector<float>yobs=ycoord;
-        int ysize=yobs.size();
-        int midpoint  = int(ysize/2);
+        if (ycoord.size()<3){
+            pair<float, float> res = make_pair(FLT_MAX, FLT_MAX);
+            return res;
+        }
+        vector<float>yobs = ycoord;
+        int ysize = yobs.size();
+        int midpoint = int(ysize/2);
         //find maximum point ( assuming it somewhere around midpoint of the yobs);
-        float ymax = max(max(yobs[midpoint], yobs[midpoint-1]),yobs[midpoint+1]);
+        float ymax = max(max(yobs[midpoint], yobs[midpoint-1]),
+                         yobs[midpoint+1]);
         float ymin = min( yobs[0], yobs[ysize-1]);
 
-        //initialize x vector, values centered around 0, forxample  -2, -1, 0, 1, 2
-        int xinit = int(ysize/2)*-1;
-        vector<float>x(ysize,0);
-        int greaterZeroCount=0;
-
-        for(int i=0; i<ysize; i++ ) {
-            x[i] = xinit+i;
-            if ( yobs[i] > ymin ) greaterZeroCount++;
-            yobs[i] = (yobs[i]-ymin)/(ymax-ymin);
-            if(yobs[i]<0) yobs[i]=0;
+        //initialize x vector, values centered around 0,
+        //forxample  -2, -1, 0, 1, 2
+        int xinit = int(ysize / 2) * -1;
+        vector<float> x(ysize, 0);
+        int greaterZeroCount = 0;
+
+        for(int i = 0; i < ysize; i++){
+            x[i] = xinit + i;
+            if (yobs[i] > ymin) greaterZeroCount++;
+            yobs[i] = (yobs[i] - ymin) / (ymax - ymin);
+            if(yobs[i] < 0) yobs[i] = 0;
         }
 
-        /*
-           cerr << "fitting yobs:" << endl; for(int i=0; i < ysize; i++ )
-           cerr << setprecision(2) << yobs[i] << ", "; cerr << endl;
-           */
-
         bool converged = false;
         int ittr = 0;
 
-        if (greaterZeroCount <= 3 ) return;
+        if (greaterZeroCount <= 3 ){
+            pair<float, float> res = make_pair(FLT_MAX, FLT_MAX);
+            return res;
+        }
         while (!converged ) {
             if ( ittr++ > 20 ) break;
-            float Rsqr=0;
-            for(int i=0; i < ysize; i++ )  { Rsqr += POW2(exp(-0.5*POW2(x[i]/s)) - yobs[i]); }
-            //       cerr << "\t\ts=" << s << " Rsqr=" << Rsqr << endl;
+            float Rsqr = 0;
+            for(int i = 0; i < ysize; i++ ){
+                Rsqr += POW2(exp(-0.5*POW2(x[i]/s)) - yobs[i]);
```

Comment:
Use C++ methods `pow` and `powf`for calculating exponents.

Reply:
Saiful, POW2 is a macro defined in mzUtils that does the square of a number.